components/tabs/src/storage.rs

```diff
@@ -399,6 +394,23 @@ fn compute_serialized_size(v: &Vec<RemoteTab>) -> usize {
     serde_json::to_string(v).unwrap_or_default().len()
 }
 
+// Similar to places/utils.js
+// handles the case where `s.is_char_boundary(max_len)`
+// is false (which would otherwise panic).
+pub fn slice_up_to(s: String, max_len: usize) -> String {
+    if max_len >= s.len() {
+        return s;
+    }
+    let mut idx = max_len;
```

Comment:
You're right on this, however since our "limit" is quite arbitrary (just to ensure we don't have insanely long titles in the payload) and these are only for local tabs to be uploaded, there wouldn't be a scenario where their constantly serialized. Though I believe to keep the code consistent I'll go with this.

Reply:
That's fine with me, just make sure to document how it works.